File: leetcode_py/Valid Parentheses.py
class Solution:
    # @return a boolean
    def isValid(self, s):
        stacks = []
        for i in s:
            if i in set(['(', '[', '{']):
                stacks.append(i)
            else:
                if len(stacks) == 0:
                    return False
                temp = stacks.pop()
                if (temp == '(' and i != ')') or (temp == '[' and i != ']') or (temp == '{' and i != '}'):
                    return False
        return len(stacks) == 0

File: leetcode_js/Valid Parentheses.js
var isValid = function(s) {
    var stack = [];
    for(var i = 0;i < s.length;i++){
        switch(s[i]){
            case '(': stack.push('(');break; 
            case '[': stack.push('[');break; 
            case '{': stack.push('{');break; 
            case ')': if(stack.length <= 0 || (stack.length > 0 &&
             stack[stack.length - 1] !== '(')) {return false;}
             stack.pop();break; 
            case ']': if(stack.length <= 0 || (stack.length > 0 &&
             stack[stack.length - 1] !== '[')) {return false;}
            stack.pop();break; 
            case '}': if(stack.length <= 0 || (stack.length > 0 &&
             stack[stack.length - 1] !== '{')) {return false;}
            stack.pop();break;  
        } 
    }
    return stack.length == 0;
};